src/ClientData/DataManager.cpp

```diff
@@ -337,4 +338,19 @@ DataManager::thread_state_change_callstack_collection() const {
   return thread_state_change_callstack_collection_;
 }
 
+void DataManager::SetSelectionTimeRange(const TimeRange& time_range) {
+  ORBIT_CHECK(std::this_thread::get_id() == main_thread_id_);
+  selection_time_range_ = time_range;
+}
+
+void DataManager::ClearSelectionTimeRange() {
+  ORBIT_CHECK(std::this_thread::get_id() == main_thread_id_);
+  selection_time_range_.reset();
+}
+
+std::optional<TimeRange> DataManager::GetSelectionTimeRange() const {
```

Comment:
Why not return by const reference? `time_range` in `OrbitApp::IsTimerActive` should be also a reference then.

Reply:
done